src/chapters/aspdotnet-model-classes/studio.rst

```diff
@@ -0,0 +1,110 @@
+   to ensure the passwords are not visible when being typed into the form.
+   You can also set ``type="email"`` on the email input, which will enable
+   some basic client-side validation. We'll tackle validation in more detail 
+   in the next studio. 
+
+Processing Form Submission
+--------------------------
+
+#. Within the ``UserController``, create an action method with this signature:
+
+   .. sourcecode:: csharp
+
+      public IActionResult AddUser(User newUser, string verify) {
+         // add form submission handling code here
+      }
+
+   This will use model binding to create a new user object, ``newUser``, and
+   pass it into your action method. 
+
+   .. admonition :: Note
+   
+      You don’t need to store the ``User`` object anywhere for this studio.
+      We’re focusing on form handling and validation in this exercise. If you
+      want to keep track of users using the method we employed in the models
+      lesson video, check out the Bonus Missions below.
+
+#. Check that the ``verify`` parameter matches the
+   password within the ``newUser`` object. If it does, render the
+   ``Index.cshtml`` view template with a message that welcomes the user by 
+   username. If the passwords don’t match, render the form again.
+
+   a. To render the ``Index.cshtml`` view with the welcome message, you need to pass in the username as part of the route for ``Redirect()``.
+      You can do so using string concatenation: ``"/user?name=" + newUser.Username``. This will pass in the username as a query parameter.
+
+   b. Passing in the value to the ``Index.cshtml`` view also requires you to modify ``Index()``. You will need to add a parameter and use ``ViewBag``.
+
+Refining Form Submission
+------------------------
+
+#. Once registered, we want the user to access the form selecting their spa services. 
+
+   a. In ``Index.cshtml``, add an ``a`` element to take the user back to the path, ``/spa``, where the ``Spa/Index.cshtml`` template will be rendered.
+
+#. If the form is re-rendered when a password is not verified, we should let the user know that their form
+   was not properly submitted. Add an ``error`` property to ``ViewBag`` letting the user know 
+   that their passwords should match. This property will need to correspond to an element in the template that will only render the error text when the passwords do not match.
+
+#. If we send a user back to re-populate the form, it would be nice to not clear their previous 
+   submission. We won't need to save the password entries in this fashion.
+   
+   a. In the form submission action method, add the ``username`` and ``email`` fields of the submitted user as 
+      ``ViewBag`` properties. 
+   
+   b. Back in the form, add a value attribute to these form fields and make them equal to the
+      ``ViewBag`` properties. 
+
+Bonus Missions
+--------------
+
+#. Add a ``Date`` property in ``User``, and initialize it to the time the
```

Comment:
I see your point about shortening the goals of the bonus mission items but this feels too short imo. Can we add something to tie this new property back into the features of the app? Like previously i think the directive was to display the user's time added in some view?